framework/moose.mk

```
@@ -12,12 +12,23 @@ ifneq (,$(findstring darwin,$(libmesh_HOST)))
 	libmesh_LDFLAGS += -headerpad_max_install_names
 endif
 
-#
-# Verify Conda
-#
-CONDA_RESULT:=$(shell bash -c "$(MOOSE_DIR)/scripts/verify_conda_libmesh.py $(MOOSE_DIR)")
-ifneq ($(CONDA_RESULT),0)
- $(error Build failed)
+#check if conda exists on the system, then run checks if it does.
+check_conda := $(shell conda list . 2> /dev/null; echo fail)
+ifneq ($(check_conda),fail)
```

```suggestion
check_conda := $(shell conda list &> /dev/null; echo $$?)
ifeq ($(check_conda),0)
```

The `.` isn't necessary here, and we want both stdout and stderr to be suppressed when this is called. I'd also like to use the explicit exit code 0 for the check rather than a string. Another exit code could indicate something is wrong with their conda installation, even if `conda list` executes.

---

No. We want stderr to be suppressed (because we don't care to parse the error) and stderr to be captured (which we need to parse).

---

We don't need to parse stdout here. We need the exit code for the process. Parsing stdout for this command comes later.

---

Yeah and we don't want to run conda list three times each time make is ran. It's slow. I talked to Nathaniel about this in person.

---

`conda list` on my machine takes 0.5s, so 1.5s or greater if slow IO. No bueno.

---

After talking offline, I understand what @loganharbour was referring to here. We should be caching the output of conda list at the same time we call it, so that we don't have to run it three times. I still would like to use some sort of exit code and not a failure string to indicate more subtle issues though. 

@socratesgorilla how do you feel about making a bash script to call here instead so that we can avoid some of the complexities of make? Logan and I both think the number of new lines could be shortened going that route. Then we might remove the old python script altogether. (As @milljm previously managed that, I would welcome his input on that last bit)

---

Feel free to remove the verify_conda_libmesh.py, as I think it serves no purpose anymore. It is to bad that interfacing with their API is slow!